clang/lib/AST/Type.cpp

```diff
@@ -385,6 +385,26 @@ void DependentBitIntType::Profile(llvm::FoldingSetNodeID &ID,
   NumBitsExpr->Profile(ID, Context, true);
 }
 
+bool BoundsAttributedType::referencesFieldDecls() const {
+  for (const auto &Decl : dependent_decls())
+    if (isa<FieldDecl>(Decl.getDecl()))
+      return true;
+  return false;
```

Comment:
```suggestion
  return llvm::any_of(dependent_decls(), [](const TypeCoupledDeclRefInfo &Info) { return isa<FieldDecl>(Info.getDecl()); });
```
Needs reformatting, but something like this should be a more clear equivalent.